corehq/apps/users/views/mobile/users.py

```diff
@@ -1126,6 +1127,60 @@ def get(self, request, domain, *args, **kwargs):
         )
 
 
+class DeleteCommCareUsers(BaseManageCommCareUserView):
+    urlname = 'delete_commcare_users'
+    page_title = ugettext_noop('Bulk Delete')
+    template_name = 'users/bulk_delete.html'
+
+    @property
+    def page_context(self):
+        context = self.main_context
+        context.update({
+            'bulk_upload_form': get_bulk_upload_form(),
+        })
+        return context
+
+    def post(self, request, *args, **kwargs):
+        try:
+            workbook = get_workbook(request.FILES.get('bulk_upload_file'))
+        except WorkbookJSONError as e:
+            messages.error(request, str(e))
+            return self.get(request, *args, **kwargs)
+
+        try:
+            sheet = workbook.get_worksheet()
+        except WorksheetNotFound:
+            messages.error(request, _("Workbook has no worksheets"))
+            return self.get(request, *args, **kwargs)
+
+        deleted_count = 0
+        usernames_with_forms = []
+        for row in sheet:
+            try:
+                username = row['username']
+            except KeyError:
+                messages.error(request, _("Please upload a file with a 'username' column."))
+                return self.get(request, *args, **kwargs)
+
+            user = CommCareUser.get_by_username(f"{username}@{request.domain}.commcarehq.org")
```

Comment:
could do a single couch request to get all users with given usernames

Reply:
413459bd50ea2d7f6fb0b706a6241ff7d6a05529